pootle/apps/virtualfolder/models.py

```diff
@@ -118,6 +202,35 @@ def clean_fields(self):
             raise ValidationError(u'The "/" location is not allowed. Use '
                                   u'"/{LANG}/{PROJ}/" instead.')
 
+    def get_adjusted_location(self, pootle_path):
+        """Return the virtual folder location adjusted to the given path.
+
+        The virtual folder location might have placeholders, which affect the
+        actual filenames since those have to be concatenated to the virtual
+        folder location.
+        """
+        count = self.location.count("/")
+
+        if pootle_path.count("/") < count:
+            raise Exception("%s is not applicable in %s" % (self, pootle_path))
```

Comment:
This is a `ValueError` to my eyes. Same with the exceptions raised below. Thoughts @jleclanche?